nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoRequestControllerFilter.java

```diff
@@ -278,11 +278,15 @@ protected void addConfiguredHeaders(HttpServletResponse response) {
         }
     }
 
-    protected void addCacheHeaders(HttpServletResponse response, boolean isPrivate, String cacheTime) {
-        String privateOrPublic = isPrivate ? "private" : "public";
-        response.setHeader("Cache-Control", privateOrPublic + ", max-age=" + cacheTime);
-        long expires = System.currentTimeMillis() + Long.parseLong(cacheTime) * 1000;
-        response.setHeader("Expires", HTTP_EXPIRES_DATE_FORMAT.format(expires));
+    protected void addCacheHeaders(HttpServletResponse response, RequestFilterConfig config) {
+        if (config.isCached()) {
+            String privateOrPublic = config.isPrivate() ? "private" : "public";
+            response.setHeader("Cache-Control", privateOrPublic + ", max-age=" + config.getCacheTime());
+            long expires = System.currentTimeMillis() + Long.parseLong(config.getCacheTime()) * 1000;
+            response.setHeader("Expires", HTTP_EXPIRES_DATE_FORMAT.format(expires));
+        } else if (config.isPrivate()) {
+            response.setHeader("Cache-Control", "private, no-cache, no-store");
```

Comment:
Why no `must-revalidate`?
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Preventing_caching it should? be present.

Reply:
Afaik `no-cache` implies `must-revalidate`. In this case concern is really `no-store` for privacy.